packages/pluggableWidgets/rating-native/src/lib/StarRating.tsx

```diff
@@ -0,0 +1,167 @@
+    static defaultProps: typeof defaultProps;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.starRef = [];
+        this.onStarButtonPress = this.onStarButtonPress.bind(this);
+    }
+
+    onStarButtonPress(rating: number) {
+        const { selectedStar } = this.props;
+
+        selectedStar && selectedStar(rating);
+    }
+
+    render() {
+        const {
+            activeOpacity,
+            animation,
+            buttonStyle,
+            containerStyle,
+            disabled,
+            emptyStar,
+            emptyStarColor,
+            fullStar,
+            fullStarColor,
+            halfStar,
+            halfStarColor,
+            halfStarEnabled,
+            iconSet,
+            maxStars,
+            rating,
+            reversed,
+            starSize,
+            starStyle
+        } = this.props;
+
+        const newContainerStyle = {
+            flexDirection: reversed ? "row-reverse" : "row",
+            justifyContent: "space-between",
+            ...StyleSheet.flatten(containerStyle)
+        } as ViewStyle;
+
+        // Round rating down to nearest .5 star
+        let starsLeft = Math.round((rating ?? DEFAULT_RATING) * 2) / 2;
+        const starButtons = [];
+
+        for (let i = 0; i < (maxStars ?? DEFAULT_MAX_RATING); i++) {
+            let starIconName = emptyStar;
+            let finalStarColor = emptyStarColor;
+
+            if (starsLeft >= 1) {
+                starIconName = fullStar;
+                finalStarColor = fullStarColor;
+            } else if (starsLeft === 0.5) {
+                starIconName = halfStar;
+                if (halfStarColor) {
+                    finalStarColor = halfStarColor;
+                } else {
+                    finalStarColor = fullStarColor;
+                }
+            }
+
+            const starButtonElement = (
+                <AnimatableView
+                    key={i}
+                    ref={node => {
+                        this.starRef.push(node);
+                    }}
+                >
+                    <StarButton
+                        activeOpacity={activeOpacity}
+                        buttonStyle={buttonStyle}
+                        disabled={disabled}
+                        halfStarEnabled={halfStarEnabled}
+                        iconSet={iconSet}
+                        onStarButtonPress={(rating: number) => {
+                            if (animation && ANIMATION_TYPES.includes(animation)) {
+                                for (let s = 0; s <= i; s++) {
+                                    const component = this.starRef[s];
+                                    if (component) {
+                                        // not sure how to correctly access prop using key
+                                        // @ts-ignore
+                                        component[animation](1000 + s * 200);
```

Comment:
If anyone can offer a suggestion on how to handle this type-error I'd appreciate some help!

Reply:
What I notice is that these animation methods are part of the AnimatableComponent type, but the refs we store for the StarRating component are of type ClassicComponent. I don't directly see a proper way to solve it :(